### PR TITLE
Update pd_inria.cpp

### PR DESCRIPTION
**Pull request by sturkmen72**
### This pullrequest changes
adds a new option to save bounding boxes as resized positive images.

### CODE REVIEW

**Comment by sturkmen72** [comment]
a screenshot of actual sample


proposed changes adds ability of saving bounding boxes like images below





the saved images almost same as the images in /INRIAPerson/96X160H96/Train/pos when default values used. also the user can create his custom sized images.

File: modules/datasets/samples/pd_inria.cpp [hunk 1]
```diff
@@ -53,14 +64,41 @@ int main(int argc, char *argv[])
...
+            int bottom = top;
+            int left = r0.x - r.x;
+            int right = left;
+
+            copyMakeBorder(img(r0), dst, top, bottom, left, right, BORDER_REFLECT);
+            resize(dst, dst, Size(rwidth, rheight), 0, 0, INTER_AREA);
+            imwrite(path + format("person_%04d.png", bboxcount++), dst);
+            }
+            else
             rectangle( img, example->bndboxes[j], Scalar( 0, 0, 255 ), 2 );
```
Comment (alalek) [inline modules/datasets/samples/pd_inria.cpp]: Please fix indentation

File: modules/datasets/samples/pd_inria.cpp [hunk 2]
```diff
@@ -53,14 +64,41 @@ int main(int argc, char *argv[])
...
         }
 
+        if (!savebbox)
+        {
```
Comment (alalek) [inline modules/datasets/samples/pd_inria.cpp]:
```
if (!savebbox)
    contrinue;  // skip UI updates
...
```

**Comment by sturkmen72** [comment]
@alalek thank you for your review and feedback.

 indentation is fixed.

"skip UI updates" no, on the contrary here we show the image with bounding boxes.

By the way there was a bug that i fixed it. ( when width of the bounding box wider )

the code may be a bit sloppy but i did not care cos it is just a sample

File: modules/datasets/samples/pd_inria.cpp (hunk 1 above)
Comment (alalek) [reply modules/datasets/samples/pd_inria.cpp]: indentation of `rectangle` statement after `else` should be fixed too

File: modules/datasets/samples/pd_inria.cpp (hunk 2 above)
Comment (alalek) [reply modules/datasets/samples/pd_inria.cpp]: > "skip UI updates" no, on the contrary here we show the image with bounding boxes.

Sure, need to invert condition:

```
if (savebbox)
    continue;  // skip UI updates
```

Or please update indentation of "if body".

File: modules/datasets/samples/pd_inria.cpp [hunk 3]
```diff
@@ -53,14 +64,47 @@ int main(int argc, char *argv[])
...
+
+            if(savebbox)
+            {
+                Mat dst;
+                int top = r0.y - r.y;
+                int bottom = top;
+                int left = r0.x - r.x;
+                int right = left;
+
+                copyMakeBorder(img(r0), dst, top, bottom, left, right, BORDER_REFLECT);
```
Comment (alalek) [inline modules/datasets/samples/pd_inria.cpp]: Why do we not just using `vpadding, vpadding, padding, padding` (without calculations with rectangles above)?
Or with scaled padding: `padding *= rwidth / r.width; vpadding *= rheight / r.height` (if applicable, it is not clear from command line description and why it is not zero by default)

**Comment by sturkmen72** [comment]
sorry again to my sloppiness :)

my main motivation was to recreate the images in `/INRIAPerson/96X160H96/Train/pos` which could not be opened with OpenCV (after updating libPNG). I was intending to use them with train_HOG.cpp. The bounding boxes in the dataset has different w/h ratios so i wanted to calculate padding.

let me try to add some explanation on the code later.

**Pull request merged**